### PR TITLE
Bertie Lax Typechecking

### PR DESCRIPTION
**Pull request by karthikbhargavan**
This PR addresses issue #86.

It requires some changes to the source code to generate better F* code. 
In particular, we work around some issues around the use of question-marks until they are resolved in hax (see https://github.com/hacspec/hax/issues/15)

It also requires some hand-edits to the generated F*, to add `let rec` for recursive functions, and to add a trait implementation for mutable `Bytes`

## Type of change

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [ ] Documentation
- [ ] Automation

## Motivation and Context

This is part of the formal verification of Bertie.

## Checklist
- [X] I have linked an issue to this PR
- [X] I have described the changes
- [ ] I have read and understood the code of conduct, contribution guidelines, and contributor license agreement
- [ ] I have added tests for all changes
- [ ] I have tested that all tests pass locally and all checks pass
- [ ] I have updated documentation if necessary

Fixes #

### CODE REVIEW

File: src/tls13formats.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ pub(crate) mod handshake_data;
 use handshake_data::{HandshakeData, HandshakeType};
 #[cfg(bench)]
 pub use handshake_data::{HandshakeData, HandshakeType};
+use tracing::dispatcher::with_default;
```
Comment (franziskuskiefer) [inline src/tls13formats.rs:27]: This seems unused?

File: proofs/fstar/bertie-lax.patch
Comment (franziskuskiefer) [inline proofs/fstar/bertie-lax.patch:1]: Why is the patch needed for lax and how is it generated?
Please ad a readme in here with explanation and instructions.

**Review by franziskuskiefer — APPROVED** [review]
Ok with the conflicts and comments addressed

File: proofs/fstar/bertie-lax.patch
Comment (karthikbhargavan) [reply proofs/fstar/bertie-lax.patch:1]: Lax checking also requires some hand-edits to the generated F*, to add `let rec` for recursive functions, and to add a trait implementation for `update_at_tc` for mutable Bytes

Comment (karthikbhargavan) [reply proofs/fstar/bertie-lax.patch:1]: We also needed to comment out some functions in tls13crypto in order to hide internal dependencies to Libcrux, these changes should go away with upcoming hax and extraction improvements.

Comment (karthikbhargavan) [reply proofs/fstar/bertie-lax.patch:1]: The patch file is the diff between the generated `extraction` folder and the hand-edited `extraction-lax` folder

Comment (franziskuskiefer) [reply proofs/fstar/bertie-lax.patch:1]: That needs to go into the readme.

**Pull request merged**